zrml/orderbook/src/tests.rs

```diff
@@ -107,367 +534,536 @@ fn it_places_orders() {
-        let share_id = Asset::CategoricalOutcome(0, 2);
         assert_ok!(Orderbook::place_order(
             RuntimeOrigin::signed(ALICE),
             market_id,
-            share_id,
-            OrderSide::Bid,
-            25,
-            10
+            maker_asset,
+            maker_amount,
+            taker_asset,
+            taker_amount,
         ));
 
-        let order_id = 0u128;
-        System::assert_last_event(
-            Event::<Runtime>::OrderPlaced {
-                order_id: 0,
-                order: Order {
-                    market_id,
-                    side: OrderSide::Bid,
-                    maker: ALICE,
-                    outcome_asset: share_id,
-                    base_asset: Asset::Ztg,
-                    outcome_asset_amount: 25,
-                    base_asset_amount: 10,
-                },
-            }
-            .into(),
-        );
+        let reserved_funds = AssetManager::reserved_balance(market.base_asset, &ALICE);
+        assert_eq!(reserved_funds, maker_amount);
 
-        let order = <Orders<Runtime>>::get(order_id).unwrap();
+        let order_id = 0u128;
+        let order = Orders::<Runtime>::get(order_id).unwrap();
         assert_eq!(
             order,
-            Order {
-                market_id,
-                side: OrderSide::Bid,
-                maker: ALICE,
-                outcome_asset: share_id,
-                base_asset: Asset::Ztg,
-                outcome_asset_amount: 25,
-                base_asset_amount: 10,
-            }
+            Order { market_id, maker: ALICE, maker_asset, maker_amount, taker_asset, taker_amount }
         );
 
+        let reserved_funds = AssetManager::reserved_balance(market.base_asset, &ALICE);
+        assert_eq!(reserved_funds, maker_amount);
+
+        assert_ok!(Orderbook::remove_order(RuntimeOrigin::signed(ALICE), order_id));
+
+        let reserved_funds = AssetManager::reserved_balance(market.base_asset, &ALICE);
+        assert_eq!(reserved_funds, 0);
+
+        assert!(Orders::<Runtime>::get(order_id).is_none());
+    });
+}
+
+#[test]
+fn removes_order_fails_if_not_market_creator() {
+    ExtBuilder::default().build().execute_with(|| {
+        let market_id = 0u128;
+        let market = market_mock::<Runtime>();
+        Markets::<Runtime>::insert(market_id, market.clone());
+
+        let maker_asset = market.base_asset;
+        let taker_asset = Asset::CategoricalOutcome(0, 2);
+
+        let taker_amount = 25 * BASE;
+        let maker_amount = 10 * BASE;
+
+        assert_ok!(Orderbook::place_order(
+            RuntimeOrigin::signed(ALICE),
+            market_id,
+            maker_asset,
+            maker_amount,
+            taker_asset,
+            taker_amount,
+        ));
+
+        let order_id = 0u128;
+
         assert_noop!(
             Orderbook::remove_order(RuntimeOrigin::signed(BOB), order_id),
             Error::<Runtime>::NotOrderCreator,
         );
+    });
+}
```

Comment:
This is already covered by `remove_order_fails_if_not_order_creator`.

```suggestion
```

Reply:
7f88460e725344dfb180576fb785fa9a737a47c8